frame/staking-rewards/src/lib.rs

```diff
@@ -453,12 +446,49 @@ pub mod pallet {
 
 		#[transactional]
 		fn extend(
-			_who: &Self::AccountId,
-			_position: Self::PositionId,
-			_amount: Self::Balance,
-			_keep_alive: bool,
+			who: &Self::AccountId,
+			position: Self::PositionId,
+			amount: Self::Balance,
+			keep_alive: bool,
 		) -> Result<Self::PositionId, DispatchError> {
-			Err("Not implemented".into())
+			let mut stake = Stakes::<T>::get(position).ok_or(Error::<T>::NoPositionFound)?;
+			let mut rewards_pool = RewardPools::<T>::try_get(stake.reward_pool_id)
+				.map_err(|_| Error::<T>::RewardsPoolNotFound)?;
+			let reward_multiplier = Perbill::one();
+
+			ensure!(
+				matches!(
+					T::Assets::can_withdraw(rewards_pool.asset_id, who, amount),
+					WithdrawConsequence::Success
+				),
+				Error::<T>::NotEnoughAssets
+			);
+
+			let boosted_amount = Self::boosted_amount(reward_multiplier, amount);
+
+			let (rewards, reductions) =
+				Self::compute_rewards_and_reductions(boosted_amount, &rewards_pool)?;
+			rewards_pool.total_shares += boosted_amount;
+			rewards_pool.rewards = rewards;
+			stake.stake += boosted_amount;
+			stake.share += boosted_amount;
```

Comment:
you use `+=` here and above
is it safe?

Reply:
Good point, this is not safe I think